### PR TITLE
usm: Split http2 from http pacakge, convert code to be golang idiomatic

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
Split http2 from http package
make packages more idiomatic

### Motivation
code cleanup, preparing for future refactor

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/http/model_linux.go [hunk 1]
```diff
@@ -123,8 +123,8 @@ func (tx *EbpfHttpTx) String() string {
 // 10 bits precision (any value will be +/- 1/1024)
 const roundMask uint64 = 1 << 10
 
-// nsTimestampToFloat converts a nanosec timestamp into a float nanosecond timestamp truncated to a fixed precision
-func nsTimestampToFloat(ns uint64) float64 {
+// NSTimestampToFloat converts a nanosec timestamp into a float nanosecond timestamp truncated to a fixed precision
+func NSTimestampToFloat(ns uint64) float64 {
```
Comment (amitslavin) [inline pkg/network/protocols/http/model_linux.go:127]: should it be exported?

**Review by amitslavin — CHANGES_REQUESTED** [review]

File: pkg/network/protocols/http/model_linux.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/protocols/http/model_linux.go:127]: yes, it is being used in http2 as well `http2/model_linux.go:46`

Comment (amitslavin) [reply pkg/network/protocols/http/model_linux.go:127]: so can you move it to common? if it is shard thro the http and http2

Comment (guyarb) [reply pkg/network/protocols/http/model_linux.go]: moved

**Review by amitslavin — APPROVED** [review]

File: pkg/network/protocols/http2/kernel_requirements.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+// MinimumKernelVersion indicates the minimum kernel version required for HTTP2 monitoring
+var MinimumKernelVersion kernel.Version
+
+func init() {
+	MinimumKernelVersion = kernel.VersionCode(5, 2, 0)
+}
+
+// Supported We only support http2 with kernel >= 5.2.0.
```
Comment (val06) [inline pkg/network/protocols/http2/kernel_requirements.go:22]: can you add here the reason why http2 supported only from 5.2.0? (is it an instructions limit?)

File: pkg/network/tracer/tracer.go [hunk 3]
```diff
@@ -136,27 +137,26 @@ func newTracer(cfg *config.Config) (*Tracer, error) {
...
+	var bpfTelemetry *nettelemetry.EBPFTelemetry
 
-	http2Supported := http.HTTP2Supported()
-	if !http2Supported && cfg.ServiceMonitoringEnabled {
-		cfg.EnableHTTP2Monitoring = false
-		log.Warnf("http2 requires a Linux kernel version of %s or higher. We detected %s", http.HTTP2MinimumKernelVersion, currKernelVersion)
-	}
+	if cfg.ServiceMonitoringEnabled {
+		if !http.Supported() {
+			errStr := fmt.Sprintf("Universal Service Monitoring (USM) requires a Linux kernel version of %s or higher. We detected %s", http.MinimumKernelVersion, currKernelVersion)
```
Comment (val06) [inline pkg/network/tracer/tracer.go:144]: make the message equivalent to the http2 case. the purpose of this refactor is "usm != http" if i understand correctly

**Review by val06 — DISMISSED** [review]
cred

File: pkg/network/protocols/http2/kernel_requirements.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/protocols/http2/kernel_requirements.go:22]: done

**Review by hmahmood — APPROVED** [review]

File: pkg/network/protocols/common.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// Copyright 2016-present Datadog, Inc.
+
+package protocols
+
+// below is copied from pkg/trace/stats/statsraw.go
+// 10 bits precision (any value will be +/- 1/1024)
+const roundMask uint64 = 1 << 10
+
+// NSTimestampToFloat converts a nanosec timestamp into a float nanosecond timestamp truncated to a fixed precision
+func NSTimestampToFloat(ns uint64) float64 {
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/common.go:13]: tried, but it does break the current function

[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/common.go:13]: discussed by heart, keeping the code as is

**Review by ISauve — APPROVED** [review]

**Review by val06 — APPROVED** [review]

**Pull request merged**